src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs

```diff
@@ -0,0 +1,199 @@
+
+                // Get the original source document again, from the latest forked solution.
+                var sourceDocument = solutionWithNewDocument.GetDocument(SemanticDocument.Document.Id);
+
+                // update source document to add partial modifiers to type chain
+                // and/or remove type declaration from original source document.
+                var solutionWithBothDocumentsUpdated = await RemoveTypeFromSourceDocumentAsync(
+                      sourceDocument).ConfigureAwait(false);
+
+                return SpecializedCollections.SingletonEnumerable(new ApplyChangesOperation(solutionWithBothDocumentsUpdated));
+            }
+
+            /// <summary>
+            /// Forks the source document, keeps required type, namespace containers
+            /// and adds it the solution.
+            /// </summary>
+            /// <param name="newDocumentId">id for the new document to be added</param>
+            /// <returns>the new solution which contains a new document with the type being moved</returns>
+            private async Task<Solution> AddNewDocumentWithSingleTypeDeclarationAndImportsAsync(
+                DocumentId newDocumentId)
+            {
+                Debug.Assert(SemanticDocument.Document.Name != State.TargetFileNameCandidate,
+                             $"New document name is same as old document name:{State.TargetFileNameCandidate}");
+
+                var root = SemanticDocument.Root;
+                var projectToBeUpdated = SemanticDocument.Document.Project;
+                var documentEditor = await DocumentEditor.CreateAsync(SemanticDocument.Document, CancellationToken).ConfigureAwait(false);
+
+                AddPartialModifiersToTypeChain(documentEditor);
+
+                // remove things that are not being moved, from the forked document.
+                var membersToRemove = GetMembersToRemove(root);
+                foreach (var member in membersToRemove)
+                {
+                    documentEditor.RemoveNode(member, SyntaxRemoveOptions.KeepNoTrivia);
+                }
+
+                var modifiedRoot = documentEditor.GetChangedRoot();
+
+                // add an empty document to solution, so that we'll have options from the right context.
+                var solutionWithNewDocument = projectToBeUpdated.Solution.AddDocument(newDocumentId, State.TargetFileNameCandidate, text: string.Empty);
+
+                // update the text for the new document
+                solutionWithNewDocument = solutionWithNewDocument.WithDocumentSyntaxRoot(newDocumentId, modifiedRoot, PreservationMode.PreserveIdentity);
+
+                // get the updated document, perform clean up like remove unused usings.
+                var newDocument = solutionWithNewDocument.GetDocument(newDocumentId);
+                newDocument = await CleanUpDocumentAsync(newDocument, CancellationToken).ConfigureAwait(false);
```

Comment:
I'm not even usign it in the method :sob: need unused param analyzer.. will use change sig and get rid of it.